src/server.js

```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+const { NodeClient } = require("hs-client");
+const NodeCache = require("node-cache");
+
+const logger = require("./logger");
+const { Resolver, ResolutionError } = require("./resolver");
+
+const hsClientOptions = {
+  network: process.env.HSD_NETWORK || "main",
+  host: process.env.HSD_HOST || "localhost",
+  port: Number(process.env.HSD_PORT) || 12037,
+  apiKey: process.env.HSD_API_KEY || "foo",
+};
+
+/**
+ * Responsible for spinning up and configuring an ExpressJS instance.
+ */
+class Server {
+  /**
+   * Create a server instance.
+   */
+  constructor() {
+    this.app = express();
+    this.resolver = new Resolver(hsClientOptions);
+    this.configure();
+  }
+
+  /**
+   * Configures routing
+   */
+  configure() {
+    this.app.get("/hnsres/:name", async (req, res) => {
+      try {
+        const result = await this.resolver.resolve(req.params.name);
+        res.json(result);
+      } catch (error) {
+        res.status(this.getStatusCodeForError(error)).send(`Handshake error: ${error.message}`);
```

Comment:
I don't think it would be possible with the current code, but I still added a `Content-Type: text/plain` header to prevent the response from being interpreted as JS, just in case :)

* 9674960